src/Ztm.WebApi/Watchers/TransactionConfirmation/IRuleRepository.cs

```diff
@@ -0,0 +1,32 @@
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using NBitcoin;
+using Ztm.WebApi.Callbacks;
+
+namespace Ztm.WebApi.Watchers.TransactionConfirmation
+{
+    public interface IRuleRepository
+    {
+        Task<Rule> AddAsync
+        (
+            uint256 transaction,
+            int confirmation,
+            TimeSpan unconfirmedWaitingTime,
+            CallbackResult successResponse,
+            CallbackResult timeoutResponse,
+            Callback callback,
+            CancellationToken cancellationToken
+        );
+
+        Task ClearCurrentWatchAsync(Guid id, CancellationToken cancellationToken);
```

Comment:
We can merge this into `UpdateCurrentWatchAsync` by allow `watchId` to be `null`.